app/controllers/stripe-setup/vat-number-company-number/vat-number/post.controller.js

```diff
@@ -12,20 +12,20 @@ const VAT_NUMBER_FIELD = 'vat-number'
 
 module.exports = (req, res) => {
   const rawVatNumber = lodash.get(req.body, VAT_NUMBER_FIELD, '')
-  const vatNumber = rawVatNumber.replace(/\s/g, '').toUpperCase()
-  const errors = validateVatNumber(vatNumber)
+  const sanitisedVatNumber = rawVatNumber.replace(/\s/g, '').toUpperCase()
+  const displayVatNumber = sanitisedVatNumber
 
+  const errors = validateVatNumber(rawVatNumber)
   if (!lodash.isEmpty(errors)) {
-    lodash.set(req, 'session.pageData.stripeSetup.vatNumber', {
-      success: false,
+    lodash.set(req, 'session.pageData.stripeSetup.vatNumberData', {
       errors: errors,
       vatNumber: rawVatNumber
     })
     return res.redirect(303, stripeSetup.vatNumber)
   } else {
-    lodash.set(req, 'session.pageData.stripeSetup.vatNumber', {
-      success: true,
-      vatNumber: vatNumber
+    lodash.set(req, 'session.pageData.stripeSetup.vatNumberData', {
```

Comment:
As discussed, I used `Data` instead of `Submission` to indicate that the session can have other data, not only form submission one (in the future) and to make it more copy&pasta friendly in case we copy the same naming conventions for future controllers which may have non-form session data